store/model.go

```diff
@@ -59,10 +62,39 @@ func newModelFromSchema(name string, schema string, s *Store) *Model {
 		schemaLoader: schemaLoader,
 		valueType:    nil,
 		store:        s,
+		indexes:      make(map[string]Index),
 	}
 	return m
 }
 
+func (m *Model) BaseKey() ds.Key {
+	return baseKey.ChildString(m.name)
+}
+
+// Indexes is a map of model properties to Indexes
+func (m *Model) Indexes() map[string]Index {
+	return m.indexes
+}
+
+// AddIndex creates a new index based on the given path string.
+// Set unique to true if you want a unique constraint on the given path.
+// See https://github.com/tidwall/gjson for documentation on the supported path structure.
+// @note: This does NOT currently build the index. If items have been added prior to adding
```

Comment:
Leaving this out as a 'future optimization'. I could see arguments for building the index if it didn't already exist, but that seems like building something we might not need yet.

Reply:
Cool, makes sense to ignore for now... again, mongodb has a nice pattern here: Let the dev run a function like, `ensureIndexes` or similar that applies index additions / changes. Def something for later.